File: frontend/libs/shared-api/src/lib/api.error.ts
export interface ApiError {
  timestamp: string;
  status: number;
  error: string;
  code: string;
  message: string;
  path: string;
  traceId?: string;
  fieldErrors?: Array<{ field: string; code: string; message: string }>;
  details?: Record<string, unknown>;
}

export function isApiError(value: unknown): value is ApiError {
  return (
    typeof value === 'object' &&
    value !== null &&
    'status' in value &&
    'code' in value &&
    'message' in value
  );
}

File: backend/shared/src/main/java/com/hisaberp/shared/web/ApiError.java
package com.hisaberp.shared.web;

import java.time.Instant;
import java.util.List;
import java.util.Map;

public record ApiError(
        Instant timestamp,
        int status,
        String error,
        String code,
        String message,
        String path,
        String traceId,
        List<FieldError> fieldErrors,
        Map<String, Object> details
) {
    public record FieldError(String field, String code, String message) {}
}
